### PR TITLE
Add a rule to avoid using Storyboards

### PR DESCRIPTION
**Pull request by turbulem**
I think we need to do this at some point.
Also if you have something to add, feel free to comment!

I think we need:
- more negative aspects of using xibs and storyboards
- some positive aspects (?)
- maybe some exceptions from the rule

### CODE REVIEW

**Comment by Wisors** [comment]
Positive:
+ WYSIWYG editor, much easier to notice an issue or warning before compilation.
+ No need to rebuild codebase, change constraint and a few seconds later see results

**Comment by leonspok** [comment]
I think, we should describe constraints for using xibs and storyboards. 

For example: 

1. It's not suggested to use other custom UI components inside xibs, except the case when this xib is fully covered with VST. Reason: if you rename this custom UI component, xib will still contain old name and it will start crashing during initialization.

**Comment by turbulem** [comment]
@Wisors 
1. I'd say that our design system efforts should make UI creation easier (and it already does) even without Xibs
2. Great point! (**will add to positives**)

@leonspok 
1. I want this rule to be only about **not using xibs at all**, with some exceptions when it's acceptable, but the guidelines for using Xibs in general should go as another rule. I feel like it's a very broad topic how to use Xibs in general

File: swift/BEST_PRACTICIES_UI.md [hunk 1]
```diff
@@ -1 +1,13 @@
 
+## Misc
+
+### Storyboards and Xibs
+
+Do not use Xibs and Storyboards in code unless you have to
```
Comment (AntonPalich) [inline swift/BEST_PRACTICIES_UI.md]: "you have to" - it's a very generic phrase. Let's think about "whitelist" of cases when you can use Xibs or Storyboards.

Comment (turbulem) [reply swift/BEST_PRACTICIES_UI.md]: I agree, but I don't know such cases. Can you please suggest a few?

Comment (AntonPalich) [reply swift/BEST_PRACTICIES_UI.md]: I don't know such cases as well :) So, maybe it's better not to leave a chance to use Xibs and Storyboards?)

Comment (turbulem) [reply swift/BEST_PRACTICIES_UI.md]: I'm thinking of rephrasing it, but I don't want it to be strict rule. Maybe someone will find a reasons to use Xibs so let's make it slightly open-ended?

"Do not use Xibs and Storyboards unless you have some strong reasons to do so and you are able to persuade other members of your team that this is a much better way"

Comment (geegaset) [reply swift/BEST_PRACTICIES_UI.md]: I'm excited :)

**Comment by dive** [comment]
My position stays the same: do not fight the system unless you have a better approach. I see no information about "better", so I voted against the proposal :)
All issues mentioned above (about naming and runtime problems) can be solved with proper tools.

File: swift/BEST_PRACTICIES_UI.md (hunk 1 above)
Comment (dive) [reply swift/BEST_PRACTICIES_UI.md]: > "Do not use Xibs and Storyboards unless you have some strong reasons to do so and you are able to persuade other members of your team that this is a much better way"

It is a much better way by default unless you have strong reasons that it is not :) I mean, let's remove these threats and provide a reasonable/flexible explanation.

**Comment by AntonPalich** [comment]
One negative Xib aspect that bothers me is that UI configuration happens in two places: Xib and code. I had "fun" time maintaining and debugging Xib-based UI, so, I can't say anything positive about it.

**Pull request closed without merging**